### PR TITLE
feat: load more departures for a specific quay

### PR DESCRIPTION
**Pull request by mortennordseth**
This PR includes new functionality for loading more departures for a specific quay. 

Closes https://github.com/AtB-AS/kundevendt/issues/9149

### CODE REVIEW

File: src/page-modules/departures/stop-place/index.tsx [hunk 1]
```diff
@@ -115,7 +137,8 @@ export function EstimatedCallList({ quay }: EstimatedCallListProps) {
             <button
               className={and(style.listItem, style.listItem__last)}
               aria-label={t(Departures.stopPlace.quaySection.a11yToQuayHint)}
-              onClick={() => alert('Not implemented yet')} // @TODO: implement function
+              onClick={getMoreDepartures}
+              disabled={isFetchingDepartures}
```
Comment (adriansberg) [inline src/page-modules/departures/stop-place/index.tsx:141]: Should it possibly be a loading state as well, so it shows that more departures are being fetched?

Comment (adriansberg) [reply src/page-modules/departures/stop-place/index.tsx:141]: Let me rephrase, since the answer to the previous question is of course that's what `isFetchingDepartures` is.

Rephrased: should we show a spinner or something that indicates that departures are being fetched? Just disabling the button might not be indicative enough. Also a11y labels.

File: src/page-modules/departures/server/journey-planner/index.ts [hunk 2]
```diff
@@ -194,6 +207,48 @@ export function createJourneyApi(
...
+
+    async estimatedCalls(input) {
+      const result = await client.query<
+        QuayEstimatedCallsQuery,
+        QuayEstimatedCallsQueryVariables
+      >({
+        query: QuayEstimatedCallsDocument,
+        variables: {
+          id: input.quayId,
+          numberOfDepartures: 10,
```
Comment (adriansberg) [inline src/page-modules/departures/server/journey-planner/index.ts:219]: Should we get as much as 10 or is 5 enough?

File: src/page-modules/departures/stop-place/index.tsx (hunk 1 above)
Comment (mortennordseth) [reply src/page-modules/departures/stop-place/index.tsx]: > should we show a spinner or something that indicates that departures are being fetched? 

Yes, I think that's better. I've updated to show a spinner while fetching.

> Also a11y labels.

Do you mean a11y label when the button is disabled? As in changing the current `aria-label` when `isFetchingDepartures` is true?

File: src/page-modules/departures/server/journey-planner/index.ts (hunk 2 above)
Comment (mortennordseth) [reply src/page-modules/departures/server/journey-planner/index.ts]: I changed it to 5. I think that's a better experience for the user 👍

File: src/page-modules/departures/stop-place/index.tsx (hunk 1 above)
Comment (mortennordseth) [reply src/page-modules/departures/stop-place/index.tsx]: Ah, you were probably thinking of `aria-disabled`. That's also fixed now.

Comment (adriansberg) [reply src/page-modules/departures/stop-place/index.tsx]: `aria-disabled` should be set to either `true` or `false` from what I can see. With a11y labels I meant that it should be clear to the user when the loading spinner is showing (and the departures are being fetched), and by only disabling the button it might not be obvious.

Comment (mortennordseth) [reply src/page-modules/departures/stop-place/index.tsx]: With the current change the `aria-disabled` is set to `true` when the state is `loading` (done by the Button component).

Comment (adriansberg) [reply src/page-modules/departures/stop-place/index.tsx]: Isn't it set to `aria-disabled={'DISABLED'}`? That's what I see. 🤔

Comment (mortennordseth) [reply src/page-modules/departures/stop-place/index.tsx]: Sorry about the confusion! I pushed some changes that I was unaware of.

File: src/pages/api/departures/journey-planner.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+        res,
+        constants.HTTP_STATUS_BAD_REQUEST,
+        ServerText.Endpoints.invalidMethod,
+      );
+    }
+
+    return tryResult(req, res, async () => {
+      return ok(
+        await client.estimatedCalls({
+          quayId: String(req.query.quayId),
```
Comment (adriansberg) [inline src/pages/api/departures/journey-planner.ts:25]: Can use `query.data` here, no?

File: src/page-modules/departures/stop-place/index.tsx [hunk 4]
```diff
@@ -55,6 +57,27 @@ type EstimatedCallListProps = {
 export function EstimatedCallList({ quay }: EstimatedCallListProps) {
   const { t } = useTranslation();
   const [isCollapsed, setIsCollapsed] = useState<boolean>(false);
+  const [departures, setDepartures] = useState<
+    DepartureData['quays'][0]['departures'][0][]
```
Comment (adriansberg) [inline src/page-modules/departures/stop-place/index.tsx:61]: There is no better type for this?

File: src/pages/api/departures/journey-planner.ts (hunk 3 above)
Comment (mortennordseth) [reply src/pages/api/departures/journey-planner.ts]: True!

**Review by adriansberg — APPROVED** [review]
